src/front/glsl_new/expressions.rs

```diff
@@ -0,0 +1,15 @@
+use super::ast::*;
+use crate::{BinaryOperator, Expression};
+
+pub fn binary_expr(
+    program: &mut Program,
```

Comment:
would it makes sense to have as a mutable method on the `Program`?

Reply:
Yeah, that could make sense. I like the idea of having functionality split into different files like: expressions, types, variables, but right now that may be a bit overkill for a single function.

Reply:
Tried it out - much simpler :D